### ISSUE TITLE
[Bug]: 14-nc886_gwas.sh; Error: Duplicate ID when using family data

### ISSUE THREAD

**Issue by ks164** [genetics-of-dna-methylation-consortium/godmc_phase2#69]
### Contact Details

ks164@duke.edu

### Scripts

14-nc886_gwas.sh

### What happened?

Script halts when loading genetic data from related individuals (MZ and DZ twins) with 'Duplicate ID' error.

The listed duplicate ID is one of my FIDs NOT an IID- most of my FIDs are duplicated due to data being from twins.

could this be due to the use of the --no-fid option?

the nc886_frequency.txt, nc886_groups.txt and nc886_scatter.jpeg files were all generated as expected.

### R version

4.3.3 (February, 2024)

### Python version

None

### Relevant log output

```shell
GoDMC2 version 1.0.0
Commit: adb482ef5c9153c2ff42e362b9f8cdfc992396c1
Commit date: 2024-10-21 17:55:15 +0100
Current time: Tue Oct 22 09:42:09 AM EDT 2024

Please ensure your scripts are up to date.
If in doubt, run 'git pull'


nc886 and clustering
Loading required package: matrixStats
Loading required package: janitor

Attaching package: ‘janitor’

The following objects are masked from ‘package:stats’:

    chisq.test, fisher.test

Warning message:
package ‘janitor’ was built under R version 4.4.0 
Loading required package: ggplot2
Saving 7 x 7 in image
GWAS intermediated vs non-methylated
PLINK v2.00a3.7LM 64-bit Intel (24 Oct 2022)   www.cog-genomics.org/plink/2.0/
(C) 2005-2022 Shaun Purcell, Christopher Chang   GNU General Public License v3
Logging to /hpc/group/sugdenlab/godmc_phase2/ERisk/results/14/intermediate_non.log.
Options in effect:
  --bfile /hpc/group/sugdenlab/godmc_phase2/ERisk/processed_data/genetic_data/data
  --ci 0.95
  --freq
  --geno-counts
  --glm allow-no-covars
  --hardy
  --maf 0.01
  --no-fid
  --out /hpc/group/sugdenlab/godmc_phase2/ERisk/results/14/intermediate_non
  --pheno /hpc/group/sugdenlab/godmc_phase2/ERisk/results/14/nc886_groups.txt
  --pheno-name intermediate_non

Start time: Tue Oct 22 09:43:09 2024
611824 MiB RAM detected; reserving 305912 MiB for main workspace.
Using 1 compute thread.
1502 samples (0 females, 0 males, 1502 ambiguous; 0 founders) loaded from
/hpc/group/sugdenlab/godmc_phase2/ERisk/processed_data/genetic_data/data.fam.
7159238 variants loaded from
/hpc/group/sugdenlab/godmc_phase2/ERisk/processed_data/genetic_data/data.bim.
Error: Duplicate ID '0 10009'.
End time: Tue Oct 22 09:43:12 2024
```

**Comment by drsora** [comment]
Hi, I'm sorry for the very late answer. You might be right about the cause of the issue, what happens if you remove the flag and rerun it? Thanks!

**Comment by ks164** [comment]
Hi,

I took the --no-fid flag off and re-ran the code, but now I have a different error:

Error: No entries in /hpc/group/sugdenlab/godmc_phase2/ERisk/results/14/nc886_groups.txt correspond to loaded sample IDs.

I have checked the nc886_groups.txt file, and the IIDs are found in my data.fam file (i.e. the IID column of nc886_groups.txt contains IDs found in the 2nd column of the data.fam file)

**Comment by drsora** [comment]
Hi,

If you give me a week or so, I can try to recreate this error in one of my cohorts with related data and get back to you on this. In the meantime, if you have any ideas on how to fix it, I'd be more than happy to hear them. Thank you for your patience!

**Comment by drsora** [comment]
Hi, 
you were correct, the columns in the fam file are swapped. Could you please either email me your fam file or run this line, which will switch the first two columns, and then rename the file data.fam to overwrite the original:
`awk '{print $2, $1, $3, $4, $5, $6}' yourfile.fam > fixed.fam`

Thank you!

**Comment by ks164** [comment]
Yes, the fam file is in standard format with FID first and IID second- the IDs in the nc886_groups.txt file correspond to the second column in my fam file.

**Comment by drsora** [comment]
Hi @ks164,

just following up, did you manage to swap the columns and did it solve the issue?
Thank you! 

> Hi, you were correct, the columns in the fam file are swapped. Could you please either email me your fam file or run this line, which will switch the first two columns, and then rename the file data.fam to overwrite the original: `awk '{print $2, $1, $3, $4, $5, $6}' yourfile.fam > fixed.fam`
> 
> Thank you!

**Comment by ks164** [comment]
sorry, I haven't had chance to re-run the code with the updated .fam file yet. I should get chance next week.

**Comment by ks164** [comment]
The module has run successfully after swapping the FID and IID.
Hopefully the results should be uploaded now, but would you mind checking since I had the same server error message as last time I tried (even though it uploaded successfully). The cohort is E-Risk.
Thanks!

**Comment by drsora** [comment]
@ks164 everything worked perfectly, I was able to download and decrypt the results. Thank you so much!
